vignettes/esApply.Rmd

```diff
@@ -0,0 +1,83 @@
+---
+title: "esApply Introduction"
+author: " Haleema Khan"
+date: October 10, 2022
+package: Biobase
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+output:
+  BiocStyle::html_document
+---
+
+# A note on `esApply` {#a-note-on-esapply .unnumbered}
+
+`ExpressionSet`s are complex objects. `exprs(ExpressionSet)` produces $G \times N$, where $G$ is the number of genes on a chip and $N$ is the number of tissues analyzed, and `pData(ExpressionSet)` produces $N \times p$, where $p$ is the number of phenotypic or demographic, etc., variables collected.
+
+Abstractly, we are often interested in evaluating functions $f(y;x)$ where $y$ is an $N$-vector of expression results for a specific gene and $x$ is an $N$-dimensional structure, coordinated with $y$, that distinguishes elements of $y$ for processing in the function $f$. A basic problem is to guarantee that the $j$th element of $y$ is correctly associated with the $j$th component of $x$.
+
+As an example, let's consider `sample.ExpressionSet`, which is an *ExpressionSet* supplied with Biobase. We will print a little report, then the first $N$-vector of gene expressions and some covariate data:
+
+```{r include=FALSE}
+devtools::load_all(".")
+load("~/Documents/Git/Biobase/data/sample.ExpressionSet.rda")
+
+```
+
+```{r}
+print(sample.ExpressionSet)
+
+```
+
+```{r}
+print(exprs(sample.ExpressionSet)[1,])
+```
+
+```{r}
+print(pData(sample.ExpressionSet)[1:2,1:3])
+```
+
+Now let's see how expressions and a covariate are related:
+
+```{r}
+print(rbind(exprs(sample.ExpressionSet[1,]),
+sex <- t(pData(sample.ExpressionSet))[1,]))
```

Comment:
I would also add spaces before line 45 to make it clear that it's still part of `rbind`.